pkg/engine/internal/planner/logical/planner.go

```diff
@@ -252,6 +256,13 @@ func buildPlanForSampleQuery(e syntax.SampleExpr, params logql.Params) (*Builder
 			}
 
 			rangeInterval = e.Left.Interval
+
+			// Check for unwrap in the LogRangeExpr
+			if e.Left.Unwrap != nil {
+				unwrapIdentifier = e.Left.Unwrap.Identifier
+				unwrapOperation = e.Left.Unwrap.Operation
+			}
```

Comment:
Is it valid for a LogQL query to have more than one `unwrap`? If so, this breaks.

Maybe in the meantime, we can return `errUnimplemented` if we see an unwrap but `unwrapIdentifier` is already set?